### PR TITLE
[Test Visibility] Document Python support for Early Flake Detection and Auto Test Retries

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
### What does this PR do? What is the motivation?

Adds Python documentation snippets for Auto Test Retries and Early Flake Detection. Also updates the features page to show parity where Python provides it.

### Merge instructions

This can merge now (`ddtrace==2.18.0` has been released).

Merge readiness:
- [x] Ready for merge

### CODE REVIEW

File: content/en/tests/flaky_test_management/auto_test_retries.md [hunk 1]
```diff
@@ -27,11 +27,45 @@ Ensure [Test Optimization][1] is configured for your test runs.
 
 ### Compatibility
```
Comment (nikita-tkachenko-datadog) [inline content/en/tests/flaky_test_management/auto_test_retries.md:28]: In preview there's some blank space between the "compatibility" and "configuration" sections that doesn't look very nice. Can we, perhaps, try moving the two headers inside the tabs

**Review by nikita-tkachenko-datadog — APPROVED** [review]

File: content/en/tests/flaky_test_management/auto_test_retries.md (hunk 1 above)
Comment (romainkomorndatadog) [reply content/en/tests/flaky_test_management/auto_test_retries.md:28]: Oh yeah, you're right. That... looks meh. I'll give it a shot.

File: content/en/tests/flaky_test_management/auto_test_retries.md [hunk 2]
```diff
@@ -120,7 +143,33 @@ Customize the Auto Test Retries with the following environment variables:
 * `DD_CIVISIBILITY_TOTAL_FLAKY_RETRY_COUNT` - a non-negative number to set the maximum total number of failed tests to retry (default: 1000)
 
 
-[1]: https://app.datadoghq.com/ci/settings/test-optimization
+[1]: https://app.datadoghq.com/ci/settings/test-service
+{{% /tab %}}
+
+{{% tab "Python" %}}
+
+<div class="alert alert-info">Auto Test Retries is available using the beta of the new pytest plugin. Set the <b>DD_PYTEST_USE_NEW_PLUGIN_BETA</b> environment variable to <b>true</b> to enable it.</div>
```
Comment (brett0000FF) [inline content/en/tests/flaky_test_management/auto_test_retries.md:151]:
```suggestion
<div class="alert alert-info">Auto Test Retries is available using the beta of the new pytest plugin. Set the <code>DD_PYTEST_USE_NEW_PLUGIN_BETA</code> environment variable to <code>true</code> to enable it.</div>
```

File: content/en/tests/flaky_test_management/auto_test_retries.md [hunk 3]
```diff
@@ -120,7 +143,33 @@ Customize the Auto Test Retries with the following environment variables:
...
+
+### Configuration
+
+After you have set up Test Optimization, you can configure Auto Test Retries from the [Test Service Settings page][1].
+
+{{< img src="continuous_integration/auto_test_retries_test_settings.png" alt="Auto Test Retries in Test Service Settings." style="width:100%" >}}
+
+The default behavior of the feature is to retry any failing test case up to 5 times. Tests that originally fail either the original setup, teardown, or fixtures in Pytest, are not retried.
+
+This behavior can be fine-tuned with the following environment variables:
```
Comment (brett0000FF) [inline content/en/tests/flaky_test_management/auto_test_retries.md:165]:
```suggestion
You can fine tune this behavior with the following environment variables:
```

File: content/en/tests/flaky_test_management/auto_test_retries.md [hunk 4]
```diff
@@ -120,7 +143,33 @@ Customize the Auto Test Retries with the following environment variables:
...
+
+* `DD_CIVISIBILITY_FLAKY_RETRY_ENABLED` - set to `0` or `false` to explicitly disable retries even if the remote setting is enabled (default: true)
```
Comment (brett0000FF) [inline content/en/tests/flaky_test_management/auto_test_retries.md:167]:
```suggestion
* `DD_CIVISIBILITY_FLAKY_RETRY_ENABLED` - set to `0` or `false` to explicitly disable retries even if the remote setting is enabled (default: `true`)
```

File: content/en/tests/flaky_test_management/early_flake_detection.md [hunk 5]
```diff
@@ -87,6 +87,14 @@ The test framework compatibility is the same as [Test Optimization Compatibility
 
 {{% /tab %}}
 
+{{% tab "Python" %}}
+
+<div class="alert alert-info">Early Flake Detection is available using the beta of the new pytest plugin. Set the <b>DD_PYTEST_USE_NEW_PLUGIN_BETA</b> environment variable to <b>true</b> to enable it.</div>
```
Comment (brett0000FF) [inline content/en/tests/flaky_test_management/early_flake_detection.md:92]:
```suggestion
<div class="alert alert-info">Early Flake Detection is available using the beta of the new pytest plugin. Set the <code>DD_PYTEST_USE_NEW_PLUGIN_BETA</code> environment variable to <code>true</code> to enable it.</div>
```

File: content/en/tests/flaky_test_management/auto_test_retries.md [hunk 6]
```diff
@@ -140,4 +189,4 @@ If you suspect there are any issues with Auto Test Retries, navigate to the [Tes
 [1]: /tests/setup/
 [2]: /tests/explorer/
 [3]: /tests/flaky_test_management/early_flake_detection
-[4]: https://app.datadoghq.com/ci/settings/test-optimization
+[4]: https://app.datadoghq.com/ci/settings/test-service
```
Comment (romainkomorndatadog) [inline content/en/tests/flaky_test_management/auto_test_retries.md:192]:
```suggestion
[4]: https://app.datadoghq.com/ci/settings/test-optimization
```

File: content/en/tests/flaky_test_management/auto_test_retries.md [hunk 7]
```diff
@@ -101,14 +118,20 @@ Customize the Auto Test Retries with the following environment variables:
 * `DD_CIVISIBILITY_TOTAL_FLAKY_RETRY_COUNT` - a non-negative number to set the maximum total number of failed tests to retry (default: 1000)
 
 
-[1]: https://app.datadoghq.com/ci/settings/test-optimization
+[1]: https://app.datadoghq.com/ci/settings/test-service
```
Comment (romainkomorndatadog) [inline content/en/tests/flaky_test_management/auto_test_retries.md:121]:
```suggestion
[1]: https://app.datadoghq.com/ci/settings/test-optimization
```

File: content/en/tests/flaky_test_management/auto_test_retries.md [hunk 8]
```diff
@@ -67,13 +70,20 @@ This behavior can be fine-tuned with the following environment variables:
 
 [jest-image-snapshot][2] is incompatible with `jest.retryTimes` unless `customSnapshotIdentifier` is passed (see [jest-image-snapshot docs][3]) to `toMatchImageSnapshot`. Therefore, auto test retries do not work unless `customSnapshotIdentifier` is used.
 
-[1]: https://app.datadoghq.com/ci/settings/test-optimization
+[1]: https://app.datadoghq.com/ci/settings/test-service
```
Comment (romainkomorndatadog) [inline content/en/tests/flaky_test_management/auto_test_retries.md:73]:
```suggestion
[1]: https://app.datadoghq.com/ci/settings/test-optimization```

File: content/en/tests/flaky_test_management/auto_test_retries.md [hunk 9]
```diff
@@ -85,11 +95,18 @@ This behavior can be fine-tuned with the following environment variables:
 * `DD_CIVISIBILITY_TOTAL_FLAKY_RETRY_COUNT` - a non-negative number to set the maximum total number of failed tests to retry (default: 1000)
 
 
-[1]: https://app.datadoghq.com/ci/settings/test-optimization
+[1]: https://app.datadoghq.com/ci/settings/test-service
```
Comment (romainkomorndatadog) [inline content/en/tests/flaky_test_management/auto_test_retries.md:98]:
```suggestion
[1]: https://app.datadoghq.com/ci/settings/test-optimization
```

File: content/en/tests/flaky_test_management/auto_test_retries.md [hunk 10]
```diff
@@ -120,7 +143,33 @@ Customize the Auto Test Retries with the following environment variables:
...
```
Comment (romainkomorndatadog) [inline content/en/tests/flaky_test_management/auto_test_retries.md:146]:
```suggestion
[1]: https://app.datadoghq.com/ci/settings/test-optimization
```

File: content/en/tests/flaky_test_management/auto_test_retries.md [hunk 11]
```diff
@@ -67,13 +70,20 @@ This behavior can be fine-tuned with the following environment variables:
 
 [jest-image-snapshot][2] is incompatible with `jest.retryTimes` unless `customSnapshotIdentifier` is passed (see [jest-image-snapshot docs][3]) to `toMatchImageSnapshot`. Therefore, auto test retries do not work unless `customSnapshotIdentifier` is used.
 
-[1]: https://app.datadoghq.com/ci/settings/test-optimization
+[1]: https://app.datadoghq.com/ci/settings/test-optimization```
```
Comment (romainkomorndatadog) [inline content/en/tests/flaky_test_management/auto_test_retries.md:73]:
```suggestion
[1]: https://app.datadoghq.com/ci/settings/test-optimization
```

File: content/en/tests/flaky_test_management/auto_test_retries.md [hunk 12]
```diff
@@ -123,6 +146,32 @@ Customize the Auto Test Retries with the following environment variables:
...
+
+The default behavior of the feature is to retry any failing test case up to five times. Tests that originally fail either the original setup, teardown, or fixtures in Pytest, are not retried.
+
+You can fine tune this behavior with the following environment variables:
+
+* `DD_CIVISIBILITY_FLAKY_RETRY_ENABLED` - set to `0` or `false` to explicitly disable retries even if the remote setting is enabled (default: `true`)
+* `DD_CIVISIBILITY_FLAKY_RETRY_COUNT` - a non-negative number to change the maximum number of retries per test case (default: `5`).
+* `DD_CIVISIBILITY_TOTAL_FLAKY_RETRY_COUNT` - a non-negative number to set the maximum total number of failed tests to retry (default: `1000`)
+
+[1]: https://app.datadoghq.com/ci/settings/test-service
```
Comment (romainkomorndatadog) [inline content/en/tests/flaky_test_management/auto_test_retries.md:171]:
```suggestion
[1]: https://app.datadoghq.com/ci/settings/test-optimization
```

**Comment by romainkomorndatadog** [comment]
I'm so sorry about the comment spam. I was trying to avoid switching context in my IDE but... it ended up being very noisy. Shameful.

**Comment by romainkomorndatadog** [comment]
@brett0000FF okay, I took all your suggestions, and fixed a link in the java Auto Test Retries page while I was at it. I think this is GTG.

**Review by brett0000FF — APPROVED** [review]

**Pull request merged**